src/mgr/QueryParams.cc

```diff
@@ -68,60 +69,61 @@ Mgr::QueryParams::find(const String& name) const
     return iter;
 }
 
-bool
-Mgr::QueryParams::ParseParam(SBuf &paramStr, Param &param)
+/// Parses the value part of a "param=value" URL section.
+/// Value can be a comma separated list of integers, or an opaque string.
+Mgr::QueryParam::Pointer
+ParseParamValue(SBuf &valueStr)
 {
-    static const CharacterSet nameChars = CharacterSet("param-name", "_") + CharacterSet::ALPHA + CharacterSet::DIGIT;
-
-    Parser::Tokenizer tok(paramStr);
-
-    SBuf name;
-    if (!tok.prefix(name, nameChars))
-        return false;
-
-    if (!tok.skip('='))
-        return false; // TODO: support name-only parameters
-
-    static const CharacterSet badValueChars = CharacterSet("param-value-bad", "&= ");
-    SBuf valueStr = tok.remaining();
-    if (valueStr.findFirstOf(badValueChars) != SBuf::npos)
-        return false;
-
-    param.first = SBufToString(name);
+    static const CharacterSet comma("comma",",");
 
+    Parser::Tokenizer tok(valueStr);
     std::vector<int> array;
     int64_t intVal = 0;
     while (tok.int64(intVal, 10, false)) {
-        static const CharacterSet comma("comma",",");
         (void)tok.skipAll(comma);
         Must(intVal >= std::numeric_limits<int>::min());
         Must(intVal <= std::numeric_limits<int>::max());
-        array.emplace_back(int(intVal));
+        array.emplace_back(intVal);
     }
 
     if (tok.atEnd())
-        param.second = new IntParam(array);
+        return new Mgr::IntParam(array);
     else
-        param.second = new StringParam(SBufToString(valueStr));
-
-    return true;
+        return new Mgr::StringParam(SBufToString(valueStr));
 }
 
-bool
+/**
+ * Syntax:
+ *   query  = param *( '&' param ) [ '#' fragment ]
```

Comment:
AFAICT, we do not require any parameters to be there:

```suggestion
 *   query  = [ param *( '&' param ) ] [ '#' fragment ]
```

Reply:
query should always have a value when received by this method. I was missing an atEnd() check.

Reply:
Fixed in 5ed02d1e6